plugins/auth-backend-module-pinniped-provider/src/authenticator.ts

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { PassportDoneCallback } from '@backstage/plugin-auth-node';
+import {
+  createOAuthAuthenticator,
+  decodeOAuthState,
+  encodeOAuthState,
+} from '@backstage/plugin-auth-node';
+import { Issuer, TokenSet, Strategy as OidcStrategy } from 'openid-client';
+
+/** @public */
+export const pinnipedAuthenticator = createOAuthAuthenticator({
+  defaultProfileTransform: async (_r, _c) => ({ profile: {} }),
+  async initialize({ callbackUrl, config }) {
+    const issuer = await Issuer.discover(
+      `${config.getString(
+        'federationDomain',
+      )}/.well-known/openid-configuration`,
+    );
+    const client = new issuer.Client({
+      access_type: 'offline', // this option must be passed to provider to receive a refresh token
+      client_id: config.getString('clientId'),
+      client_secret: config.getString('clientSecret'),
+      redirect_uris: [callbackUrl],
+      response_types: ['code'],
+      scope: config.getOptionalString('scope') || '',
+      id_token_signed_response_alg: 'ES256',
```

Comment:
@cfryanr does the Pinniped Supervisor ever sign tokens with a different algorithm than this?

Reply:
Thanks @cfryanr -- I think I'm comfortable leaving it hardcoded here for the time being then.